plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiClassBuilderInterceptor.kt

```diff
@@ -110,6 +118,32 @@ class JvmAbiClassBuilderInterceptor(
             delegate.visitEnclosingMethod(owner, name, desc)
         }
 
+        override fun newField(
+            declaration: IrField?, access: Int, name: String, desc: String, signature: String?, value: Any?
+        ): FieldVisitor {
+            if (keepClassAsIs || removeClassFromAbi) {
+                // We don't care about fields when we remove or keep this class completely.
+                return delegate.newField(declaration, access, name, desc, signature, value)
+            }
```

Comment:
Nevertheless what `keepClassAsIs` says in the original implementation regarding of the value of the flag all `private` fields were deleted unconditionally. I kept that logic here.